ci-operator/step-registry/cucushift/chainupgrade/toimage/cucushift-chainupgrade-toimage-commands.sh

```diff
@@ -943,14 +943,14 @@ function run_upgrade_e2e() {
     parallel_cucumber -n "${PARALLEL}" --first-is-1 --type cucumber --serialize-stdout --combine-stderr --prefix-output-with-test-env-number --exec \
     'export OPENSHIFT_ENV_OCP4_USER_MANAGER_USERS=$(echo ${USERS} | cut -d "," -f ${TEST_ENV_NUMBER},$((${TEST_ENV_NUMBER}+${PARALLEL})),$((${TEST_ENV_NUMBER}+${PARALLEL}*2)));
      export WORKSPACE=/tmp/dir${TEST_ENV_NUMBER};
-     parallel_cucumber --group-by found --only-group ${TEST_ENV_NUMBER} -o "--tags \"${E2E_RUN_TAGS} and not @admin\" -p junit"' || true
+     parallel_cucumber --group-by found --only-group ${TEST_ENV_NUMBER} -o "--tags \"${E2E_RUN_TAGS} and not @admin\" -p junit"' &>> "${ARTIFACT_DIR}/e2etest-log.txt" || true
 
     # run admin tests
     export BUSHSLICER_REPORT_DIR="${ARTIFACT_DIR}/parallel-admin-${idx}"
     parallel_cucumber -n "${PARALLEL}" --first-is-1 --type cucumber --serialize-stdout --combine-stderr --prefix-output-with-test-env-number --exec \
     'export OPENSHIFT_ENV_OCP4_USER_MANAGER_USERS=$(echo ${USERS} | cut -d "," -f ${TEST_ENV_NUMBER},$((${TEST_ENV_NUMBER}+${PARALLEL})),$((${TEST_ENV_NUMBER}+${PARALLEL}*2)));
      export WORKSPACE=/tmp/dir${TEST_ENV_NUMBER};
-     parallel_cucumber --group-by found --only-group ${TEST_ENV_NUMBER} -o "--tags \"${E2E_RUN_TAGS} and @admin\" -p junit"' || true
+     parallel_cucumber --group-by found --only-group ${TEST_ENV_NUMBER} -o "--tags \"${E2E_RUN_TAGS} and @admin\" -p junit"' >> "${ARTIFACT_DIR}/e2etest-log.txt" || true
```

Comment:
This is the execution log. May i know why we write it into ARTIFACT_DIR ?

Reply:
now all e2e test logs are kept in [artifact dir](https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/54558/rehearse-54558-periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-4.12-upgrade-from-eus-4.10-azure-ipi-f28/1815196465195126784/artifacts/azure-ipi-f28/cucushift-chainupgrade-toimage/artifacts/), so I assume it's better to save them together for readers checking it quickly and conveniently in one place.

Reply:
The others are report files, except `e2etest-log.txt` created by this PR.
Execution log files are outside of artifact_dir.

Reply:
ok, could u suggest a suitable dir for the execution log? actually the pr is to split it from upgrade log, but not care too much about the log dir.

Reply:
How about the same dir with upgrade log, but a different name ?

Reply:
do u mean the same dir with `build.log`, right? I'm not sure if the permission is available for the step dir, which looks like manged by prow itself. but i can have a try.

Reply:
From the [test](https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/54558/rehearse-54558-periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-4.12-upgrade-from-eus-4.10-azure-ipi-f28/1815315951055278080/artifacts/azure-ipi-f28/cucushift-chainupgrade-toimage/) , outputting logs without specifying an absolute path does not work. I just check the [doc](https://docs.ci.openshift.org/docs/architecture/step-registry/#available-environment-variables) to find available env vars for a step, it seems `{ARTIFACT_DIR}` is more suitable for the purpose here. Do u know what's the env var name for the suggested dir?

Reply:
Let's see if `${ARTIFACT_DIR}/../filename` works ?

Reply:
it does not work, i'm afraid it's not the recommended dir for saving logs.

Reply:
Add a sleep here and let's check in the running cluster, e.g, check all the ENVs and see if we can find a proper one

Reply:
I doubt even there are other envs, but not all of them are suitable for keeping logs, and maybe no one is more better than artifact. Except for the distinguish between report files and execution logs, do u know any other harm to leave it in artifacts. Actually, I also see the execution logs are kept in artifacts together with junit files(for example, ginkgo [test](https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/54558/rehearse-54558-periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-4.12-upgrade-from-eus-4.10-azure-ipi-f28/1815716027355369472/artifacts/azure-ipi-f28/openshift-extended-upgrade-post/artifacts/extended.log)), so i think we could choose the same way here, wdyt?

Reply:
cc @jianlinliu any opinion here?

Reply:
I am okay with keeping the output log in artifact, @liangxia what is your concern for that ?

Reply:
We are parsing files in artifact for the reports.  If files are added there without cation, it may break the result parsing.

Reply:
hmm, does the parsing job watch part of the files with particular postfix or all files? I see all reports files are named with `*.yaml` and `*.xml`, the e2e log is named with `*.txt`, could it be skipped from name postfix?

Reply:
How about use a specific folder for this kind of usage ? So that we can skip this folder, and we can put other files there too if we have the need in the future.

Reply:
sgtm, test [result](https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/54558/rehearse-54558-periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-4.12-upgrade-from-eus-4.10-azure-ipi-f28/1815967364601090048/artifacts/azure-ipi-f28/cucushift-chainupgrade-toimage/artifacts/)